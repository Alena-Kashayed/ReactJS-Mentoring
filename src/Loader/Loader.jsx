import React from 'react';
import classNames from 'classnames/bind';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import styles from './Loader.scss';

const Loader = ({ status }) => {
  const cx = classNames.bind(styles);
  return (
    status ?
      <div className={styles.loader}>
        <div className={styles.preloader1}>
          <div>Loading</div>
          <span className={cx('line', 'line1')} />
          <span className={cx('line', 'line2')} />
          <span className={cx('line', 'line3')} />
          <span className={cx('line', 'line4')} />
          <span className={cx('line', 'line5')} />
          <span className={cx('line', 'line6')} />
          <span className={cx('line', 'line7')} />
          <span className={cx('line', 'line8')} />
          <span className={cx('line', 'line9')} />
        </div>
      </div> : null
  );
};

Loader.propTypes = {
  status: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  status: state.loader.status,
});

export default connect(mapStateToProps)(Loader);
