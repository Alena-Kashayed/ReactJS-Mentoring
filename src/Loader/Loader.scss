.loader {
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  z-index: 3;
  background-color: rgba(0, 0, 0, 0.5);

  .preloader {
    width: 66px;
    height: 12px;
  }

  div {
    color: #fff;
    margin: 5px 0;
    text-transform: uppercase;
    text-align: center;
    font-family: 'Arial', sans-serif;
    font-size: 10px;
    letter-spacing: 2px;
  }

  .preloader .line {
    width: 1px;
    height: 12px;
    background: #fff;
    margin: 0 1px;
    display: inline-block;
    animation: opacity-1 1000ms infinite ease-in-out;
  }

  .preloader .line1 { animation-delay: 800ms; }
  .preloader .line2 { animation-delay: 600ms; }
  .preloader .line3 { animation-delay: 400ms; }
  .preloader .line4 { animation-delay: 200ms; }
  .preloader .line6 { animation-delay: 200ms; }
  .preloader .line7 { animation-delay: 400ms; }
  .preloader .line8 { animation-delay: 600ms; }
  .preloader .line9 { animation-delay: 800ms; }

  @keyframes opacity-1 {
    0% {
      opacity: 1;
    }
    50% {
      opacity: 0;
    }
    100% {
      opacity: 1;
    }
  }
}
