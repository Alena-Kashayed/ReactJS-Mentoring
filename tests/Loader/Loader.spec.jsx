import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import Loader from '../../src/Loader/Loader';

const mockStore = configureStore();

describe('Loader', () => {
  it('is shown', () => {
    const wrapper = shallow(<Loader
      store={mockStore({
        loader: { status: true },
      })}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('is hidden', () => {
    const wrapper = shallow(<Loader
      store={mockStore({
        loader: { status: false },
      })}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
