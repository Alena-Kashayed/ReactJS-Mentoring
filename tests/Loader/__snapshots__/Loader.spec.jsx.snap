// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader is hidden 1`] = `
<Loader
  dispatch={[Function]}
  status={false}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`Loader is shown 1`] = `
<Loader
  dispatch={[Function]}
  status={true}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
